cli/generator-op-microservice/generators/microservice/templates/gql.service.ts

```diff
@@ -0,0 +1,88 @@
+import express from 'express';
+import { ApolloServer } from 'apollo-server-express';
+import fs from 'fs';
+import https from 'https';
+import http from 'http';
+import { mergeSchemas } from 'graphql-tools';
+const { ApolloLogExtension } = require('apollo-log');
+<% if (dbSupport) { -%>
+import mongoose from 'mongoose';
+<% } -%>
+
+import gqlSchema from './src/typedef.graphql';
+import { <%= resolverName %> as resolver } from './src/resolver';
+
+/* Setting port for the server */
+const port = process.env.PORT || 8080;
+const environment = process.env.NODE_ENV;
+
+const app = express();
+const extensions = [() => new ApolloLogExtension({
+  level: 'info',
+  timestamp: true,
+})];
+
+<% if (dbSupport) { -%>
+    /* Configuring Mongoose */
+    mongoose.plugin((schema: any) => { schema.options.usePushEach = true; });
+    mongoose.set('useNewUrlParser', true);
+    mongoose.set('useFindAndModify', false);
+    mongoose.set('useCreateIndex', true);
+    mongoose.set('useUnifiedTopology', true);
+
+    /* Establishing mongodb connection */
+    const dbCredentials = (process.env.DB_USER && process.env.DB_PASSWORD)
+      ? `${process.env.DB_USER}:${process.env.DB_PASSWORD}@`
+      : '';
+    const dbConnection = `mongodb://${dbCredentials}${process.env.DB_PATH}/${process.env.DB_NAME}`;
+
+    mongoose.connect(dbConnection, { useNewUrlParser: true, useCreateIndex: true }).catch(console.error);
+
+    mongoose.connection.on('error', error => {
+      console.error(error);
+    });
+<% } -%>
+
+/* Defining the Apollo Server */
+const apollo = new ApolloServer({
+  playground: environment !== 'production',
```

Comment:
Can you use the complete `process.env.NODE_ENV` instead of a separate variable, as the full version is already being used at other places for DB username, password and other things.

Reply:
Hi, I am using that already. please see line no 17

https://github.com/1-Platform/one-platform/blob/ac04662147102e6c0180ae0a17f386231dfa2bce/cli/generator-op-microservice/generators/microservice/templates/gql.service.ts#L17